### PR TITLE
new: bold friends in tab list

### PR DESCRIPTION
**Pull request by osbourn**
Changes the formatting of friends in the tab menu to be bold.

The first time the player opens the tab menu, a thread will be started that sends a request to https://api.sk1er.club/friends/(playerUUID) to find a list of the friends. It will be stored in a List object.
From there, every time a player views a name in the tab menu, it will convert it to a username and check to see if it is in the friends list. If it is, the bold color code is added.

### CODE REVIEW

File: src/main/java/club/sk1er/hytilities/handlers/lobby/tab/TabChanger.java [hunk 1]
```diff
@@ -50,6 +60,48 @@ public static String modifyName(String name) {
...
+                            name = name.substring(0, firstPlus) + "\u00a7l" + name.substring(firstPlus);
+                            int firstBracket = name.indexOf("]");
+                            name = name.substring(0, firstBracket) + "\u00a7l" + name.substring(firstBracket);
+                        }
+                    }
+                } else {
+                    // Only start a thread to retrieve the list of usernames if one does not already exist
+                    if (fetchFriendInfoThread == null) {
+                        // Create and start a new thread to fetch friend info
+                        fetchFriendInfoThread = new Thread(() -> {
```
Comment (asbyth) [inline src/main/java/club/sk1er/hytilities/handlers/lobby/tab/TabChanger.java:97]: would likely be better to use `Multithreading.runAsync(runnable)`

File: src/main/java/club/sk1er/hytilities/handlers/lobby/tab/TabChanger.java [hunk 2]
```diff
@@ -74,4 +126,60 @@ private static boolean isSkyblockTabInformationEntry(NetworkPlayerInfo networkPl
...
+            connection.setUseCaches(true);
+            connection.addRequestProperty("User-Agent", "Mozilla/4.76 (Hytilities Mod V1.0)");
+
+            // Connection will timeout after 15 seconds
+            connection.setReadTimeout(15000);
+            connection.setConnectTimeout(15000);
+
+            // Read output from connection
+            connection.setDoOutput(true);
+            InputStream is = connection.getInputStream();
```
Comment (asbyth) [inline src/main/java/club/sk1er/hytilities/handlers/lobby/tab/TabChanger.java:166]: unclosed stream? can use `try (InputStream is = connection.getInputStream()) { // code }`, unless if this is done for another reason that i'm missing

**Comment by kazooboi** [comment]
Would this have an impact on network or server? Sorry I'm new to modding

**Comment by asbyth** [comment]
> Would this have an impact on network or server? Sorry I'm new to modding

no, it wouldn't have any sort of impact

**Comment by osbourn** [comment]
Superseded by #101

**Pull request closed without merging**